lib/bumblebee/text/question_answering.ex

```diff
@@ -0,0 +1,172 @@
+    Shared.validate_architecture!(spec, :for_question_answering)
+
+    opts =
+      Keyword.validate!(opts, [
+        :compile,
+        doc_stride: 128,
+        top_k: 1,
+        defn_options: []
+      ])
+
+    top_k = opts[:top_k]
+    compile = opts[:compile]
+    defn_options = opts[:defn_options]
+    doc_stride = opts[:doc_stride]
+
+    batch_size = compile[:batch_size]
+    sequence_length = compile[:sequence_length]
+
+    if compile != nil and (batch_size == nil or sequence_length == nil) do
+      raise ArgumentError,
+            "expected :compile to be a keyword list specifying :batch_size and :sequence_length, got: #{inspect(compile)}"
+    end
+
+    {_init_fun, predict_fun} = Axon.build(model)
+
+    scores_fun = fn params, input ->
+      # input = Utils.Nx.composite_flatten_batch(input)
+      output = predict_fun.(params, input)
+    end
+
+    Nx.Serving.new(
+      fn ->
+        predict_fun =
+          Shared.compile_or_jit(scores_fun, defn_options, compile != nil, fn ->
+            inputs = %{
+              "input_ids" => Nx.template({batch_size, sequence_length}, :s64),
+              "attention_mask" => Nx.template({batch_size, sequence_length}, :s64)
+            }
+
+            [params, inputs]
+          end)
+
+        fn inputs ->
+          inputs = Shared.maybe_pad(inputs, batch_size)
+
+          scores = predict_fun.(params, inputs)
+        end
+      end,
+      batch_size: batch_size
+    )
+    |> Nx.Serving.client_preprocessing(fn input ->
+      {inputs, multi?} =
+        Shared.validate_serving_input!(input, fn
+          %{question: question, context: context}
+          when is_binary(question) and is_binary(context) ->
+            {:ok, {question, context}}
+
+          other ->
+            {:error,
+             "expected input map with :question and :context keys, got: #{inspect(other)}"}
+        end)
+
+      all_inputs =
+        Bumblebee.apply_tokenizer(tokenizer, inputs,
+          length: sequence_length,
+          return_special_tokens_mask: true,
+          return_offsets: true
+        )
+
+      inputs = Map.take(all_inputs, ["input_ids", "attention_mask"])
+      {Nx.Batch.concatenate([inputs]), {all_inputs, multi?}}
+    end)
+    |> Nx.Serving.client_postprocessing(fn outputs, metadata, {inputs, multi?} ->
+      %{
+        results:
+          Enum.zip_with(
+            Utils.Nx.batch_to_list(inputs),
+            Utils.Nx.batch_to_list(outputs),
+            fn inputs, outputs ->
+              answer_start_index = inputs["start_offsets"] |> Nx.argmax() |> Nx.to_number()
+
+              answer_end_index = inputs["end_offsets"] |> Nx.argmax() |> Nx.to_number()
+
+              answer_tokens = inputs["input_ids"][answer_start_index..answer_end_index]
+
+              answers = Bumblebee.Tokenizer.decode(tokenizer, answer_tokens)
+
+              %{
+                text: answers,
+                start: inputs["start_offsets"],
```

Comment:
```suggestion
                start: Nx.to_number(inputs["start_offsets"][answer_start_index]),
```

something like this, except I would move it to a separate variable :)

Reply:
Alirght last question I hope, where does the score come from ?

Reply:
The model returns start/end logits, but we need to normalize it to 0-1 with softmax, you can compute start_scores and end_scores with `Axon.Activations.softmax`, then I think the score is just start score * end score for the argmax index.